includes/checkout/template.php

```diff
@@ -1099,5 +1099,23 @@ function edd_receipt_show_download_files( $item_id, $receipt_args, $item = array
 		}
 	}
 
-	return apply_filters( 'edd_receipt_show_download_files', $ret, $item_id, $receipt_args, $item );
+	if ( has_filter( 'edd_receipt_show_download_files' ) ) {
+		$cart = array();
+		if ( ! empty( $item->order_id ) ) {
+			$order = edd_get_order_by( 'id', $item->order_id );
+			$cart  = edd_get_payment_meta_cart_details( $order->id, true );
+		}
+		$ret = apply_filters( 'edd_receipt_show_download_files', $ret, $item_id, $receipt_args, $cart[ $item->cart_index ] );
+	}
+
+	/**
+	 * Modifies whether the receipt should show download files.
+	 *
+	 * @since 3.0
+	 * @param bool                   $ret          True if the download files should be shown.
+	 * @param int                    $item_id      The download ID.
+	 * @param array                  $receipt_args Args specified in the [edd_receipt] shortcode.
+	 * @param \EDD\Orders\Order_Item $item        The order item object.
+	 */
+	return apply_filters( 'edd_order_receipt_show_download_files', $ret, $item_id, $receipt_args, $item );
```

Comment:
Similar to what I said above, we need to account for `$item` being an array here instead of an `Order_Item` object. If it is an array then we need to turn it into an `Order_Item`.